include/xlnt/utils/date.hpp

```diff
@@ -49,21 +50,54 @@ struct XLNT_API date
     static date from_number(int days_since_base_year, calendar base_date);
 
     /// <summary>
-    /// Constructs a data from a given year, month, and day.
+    /// Constructs a date from a given year, month, and day.
     /// </summary>
     date(int year_, int month_, int day_);
 
     /// <summary>
-    /// Return the number of days between this date and base_date.
+    /// Constructs an empty date (a call to is_null() will return true).
+    /// </summary>
+    date() = default;
+
+    /// <summary>
+    /// Return the number of days between this date and base_date. The date could be in an empty/invalid state, so you may want to call is_null() first!
+    /// If this function is called when having an empty/invalid state, an exception will be thrown.
     /// </summary>
     int to_number(calendar base_date) const;
 
     /// <summary>
     /// Calculates and returns the day of the week that this date represents in the range
     /// 0 to 6 where 0 represents Sunday.
+    /// Returns -1 if the weekday could not be determined.
     /// </summary>
     int weekday() const;
 
+    /// <summary>
+    /// Returns the year of the date. The date could be in an empty/invalid state, so you may want to call is_null() first!
+    /// If this function is called when having an empty/invalid state, an exception will be thrown.
```

Comment:
Should we specify which exception will be thrown or is that an implementation detail?

Reply:
Usually I would say no, because the programmer should call `is_null()` first, instead of relying on exception handling. However, this is just how I personally see it, and other people might have other preferences. Due to that, I changed the documentation to specify the exact exception being thrown.